### PR TITLE
Migration to Python 3

### PR DESCRIPTION
**Pull request by joke325**
issue #812

### CODE REVIEW

**Comment by ronaldtse** [comment]
@joke325 you have merge rights now, sorry forgot to you add you! 😉

File: src/tests/CMakeLists.txt [hunk 1]
```diff
@@ -28,7 +28,7 @@ find_package(JSON-C 0.11 REQUIRED)
 # Note that we do this call early because Google Test will also do
 # this but with less strict version requirements, which will cause
 # problems for us.
-find_package(PythonInterp 2.7 EXACT)
+find_package(Python3 COMPONENTS Interpreter)
```
Comment (dewyatt) [inline src/tests/CMakeLists.txt]: We'll need cmake 3.12+ for this.

```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index edf3ff12..c57448bf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -27,7 +27,8 @@
 # 3.10+ for CPackFreeBSD (FreeBSD packaging)
 # 3.10+ for gtest_discover_tests (parallel rnp_tests)
 # 3.12+ for NAMELINK_COMPONENT (for better RPM packaging)
-cmake_minimum_required(VERSION 3.10)
+# 3.12+ for Python3 find module
+cmake_minimum_required(VERSION 3.12)
 
 # contact email, other info
 include(cmake/info.cmake)
```

**Comment by ronaldtse** [comment]
@dewyatt @joke325 this seems ready to go?

**Comment by joke325** [comment]
@ronaldtse it's waiting for review.

**Comment by ni4** [comment]
@ronaldtse @joke325 I think we should first fix windows CI (or wait till it fixes itself) before merging this.

**Comment by ronaldtse** [comment]
Not a python guy but looks reasonable to me? All tests are passing. @dewyatt @ni4?

**Comment by ni4** [comment]
@ronaldtse Now, when we are sure that Windows build passes, I can approve this. Thanks, really step forwards from outdated Python 2.7!

**Comment by dewyatt** [comment]
Thanks @joke325!

**Pull request merged**